aten/src/TH/generic/THTensor.cpp

```diff
@@ -438,20 +438,24 @@ void THTensor_(squeeze)(THTensor *self, THTensor *src)
 
   THTensor_(set)(self, src);
 
+  std::vector<int64_t> newSize;
+  std::vector<int64_t> newStride;
   for(d = 0; d < src->dim(); d++)
   {
     if(src->size(d) != 1)
     {
-      if(d != ndim)
-      {
-        self->set_size(ndim, src->size(d));
-        self->set_stride(ndim, src->stride(d));
+      if (d != ndim) {
+        newSize.push_back(src->size(d));
+        newStride.push_back(src->stride(d));
+      } else {
+        newSize.push_back(self->size(ndim));
+        newStride.push_back(self->stride(ndim));
```

Comment:
This conditional is very confusing. If `d != ndim`, you use d. But if `d == ndim`, you use ndim. But it's equivalent to use d. You don't need the if statement.

Reply:
I think if statement is still needed. If `d != ndim`, the size/stride of `src` is used, while if `d == ndim`, the size/stride of `self` is used.